src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,64 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_VALUE = 10;
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(Random random, ColorSupplier colorSupplier) {
+        this.random = random;
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figures getRandomFigure() {
+        int randomInt = random.nextInt(Figure.values().length);
+
+        switch (Figure.values()[randomInt]) {
+            case CIRCLE:
+                return getCircle();
+            case RECTANGLE:
+                return getRectangle();
+            case SQUARE:
+                return getSquare();
+            case RIGHTTRIANGLE:
+                return getRightTriangle();
+            default :
+                return getIsoscelesTrapezoid();
+        }
+    }
+
+    private Circle getCircle() {
+        return new Circle(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_RANDOM_VALUE));
+    }
+
+    private Square getSquare() {
+        return new Square(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_RANDOM_VALUE));
+    }
+
+    private RightTriangle getRightTriangle() {
+        return new RightTriangle(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_RANDOM_VALUE),
+                random.nextInt(MAX_RANDOM_VALUE));
+    }
+
+    private Rectangle getRectangle() {
+        return new Rectangle(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_RANDOM_VALUE),
+                random.nextInt(MAX_RANDOM_VALUE));
+    }
+
+    private IsoscelesTrapezoid getIsoscelesTrapezoid() {
```

Comment:
please move private methods below 2 public methods
public methods should go first